NorthstarDedicatedTest/voicebasic.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <sapi.h>
+#include <iostream>
+#include <string>
+using namespace std;
+class BasicVoice
+{
+  protected:
+	ISpVoice* pVoice;
+	HRESULT hr, a;
+
+  public:
+	float speechRate = 1.0;
+	float speechVolume = 1.0;
+	BasicVoice()
+	{
+		pVoice = NULL;
+		a = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED);
+		if (!FAILED(a))
+		{
+			hr = CoCreateInstance(CLSID_SpVoice, NULL, CLSCTX_ALL, IID_ISpVoice, (void**)&pVoice);
+		}
+	}
+	virtual void speakText(char* input);
+	virtual void setRate(float rate);
+	virtual void setVolume(float volume);
+	virtual void outSpeech();
+	virtual void skipSpeech();
+	virtual ~BasicVoice()
+	{
+		::CoUninitialize();
+		delete pVoice;
```

Comment:
Why are you deleting pVoice? COM objects are managed with the reference counting from the release function.